lib/oli/delivery/attempts/page_lifecycle/graded.ex

```diff
@@ -164,11 +162,12 @@ defmodule Oli.Delivery.Attempts.PageLifecycle.Graded do
           end
 
         %ResourceAttempt{lifecycle_state: :submitted, resource_access_id: resource_access_id} ->
+          {:ok, resource_access} = Oli.Delivery.Metrics.mark_progress_completed(Oli.Repo.get(ResourceAccess, resource_access_id))
```

Comment:
Stylistically, Elixir code should generally pipe results of functions instead of nesting: function calls:

```
{:ok, resource_access} = Oli.Repo.get(ResurceAccess, resource_access_id)
|> Oli.Delivery.mark_progress_completed()
```